### DISCUSSION TITLE
Support for Concurrent Users

### DISCUSSION THREAD

**Discussion by spammenotinoz** [open-webui/open-webui#4378] [General]
Currently with two users, the system will only complete requests sequentially.  This is the case even when users are using different LiteLLM model providers, and particularly evident when 1 of the 2 users is generating pictures.

Would like the ability for the system to handle parallel API requests to different and\or the same provider.

Have considered deploying Open-WebUI locally for both users or load balancing Open-WebUI itself, but seems extreme for 2 users.

From the comments I hear people with multiple users, wondering if I did something really dumb in the setup.
PS: Latest Image  "docker.io/open-webui/open-webui:latest", but this issue has always existed for me, so perhaps user error.

**Comment by gaby** [comment]
@spammenotinoz Is this still the case? Did you find a solution?

**Reply by spammenotinoz** [reply]
> @spammenotinoz Is this still the case? Did you find a solution?

My primary issue was resolved a long-time ago, turned out that all chats paused during image generation, that was fixed a while ago.
Minor issue that some Functions appear sequential\single use... only because an issue recently when I started to use Deek-Seek function to visualise the thinking tokens..

**Reply by gaby** [reply]
Thanks for the update! I don't have image generation enabled, so maybe that's why I haven't notice the slow down.

**Comment by tabernario** [comment]
I have some issue with something related.
Sorry if this is not the place but I have a problem and I can't figure it out.
I can't do 2 concurrent request using Openwebui, I tried many times and always does first one of the request and then does the another one. This happens with same user but also with different users.
The strange part is that if I make the curl requests to Ollama directly it respond simultaneously but when I try to do it in openwebui it respond sequentially even though I try to do it with /ollama/api/generate which allegedly is "transparent passthrough to the native Ollama API via a proxy route."
So it seems that may be there's some parameter in openwebui that I should set to have the possibility of parallel request.
Openwebui and ollama are in different computers and they work fine in a sequential job it just doesn't work in parallel, which is annoying since one user has to be waiting to the other to finish and sometimes it's not operational.
I was searching for information regarding this matter but this is the closest subjet I could find
I would be grateful for any help.
